### PR TITLE
Add protocol docu to the help docs

### PR DESCRIPTION
**Pull request by rabuzarus**
There is now a page in the help docs with protocol documentation

Additionally:
* Protocol spec was updated (history and changed outdated link)
* Friendica connection schema was added to `/spec`
* Restructure of some points in the help doc main page (some points belong rather to developer documentation)
* some frio css styling for the help docs

### CODE REVIEW

**Comment by rabuzarus** [comment]
BTW: I saw that you have a copy of the salmon specs. I think it would be good to have this online as webpage. If we use it developers should have an easy access to it.

We have the same problem with the specs for portable contacts. I added new links (in the wiki and in the docu pages) to archive.org but to have as webpage which is searchable with google (or other) would be good for developers outside of the friendica universum

File: doc/Protocol.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+* [Document with the DFRN specification](spec/dfrn2.pdf)
+* [Schema of the contact request process](spec/dfrn2_contact_request.png)
+* [Schema of the contact request confirmation](spec/dfrn2_contact_confirmation.png)
+* [Description of the message flow](help/Message-Flow)
+
+ActivityStreams
+---
+
+Friendica is using ActivityStreams in version 1.0 for its activities and object types. Additional types are using for non standard activities.
```
Comment (tobiasd) [inline doc/Protocol.md]: Could you add a line break after the "." we try to keep that rule for the docs except for lists.

Comment (rabuzarus) [inline doc/Protocol.md]: Can we write this rules down somewhere? It's hard to remember the docu rules when someone don't work very often at the documentation part

Comment (tobiasd) [inline doc/Protocol.md]: It's the only rule, to avoid long sentences and make diffs easier ;--) but yes, you can write it down somewhere.

Comment (rabuzarus) [inline doc/Protocol.md]: Ahhh OK :-)

File: doc/Protocol.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+* [Document with the DFRN specification](spec/dfrn2.pdf)
+* [Schema of the contact request process](spec/dfrn2_contact_request.png)
+* [Schema of the contact request confirmation](spec/dfrn2_contact_confirmation.png)
+* [Description of the message flow](help/Message-Flow)
+
+ActivityStreams
+---
+
+Friendica is using ActivityStreams in version 1.0 for its activities and object types.
+Additional types are using for non standard activities.
```
Comment (annando) [inline doc/Protocol.md]: "are used"

File: include/dfrn.php [hunk 3]
```diff
@@ -3,7 +3,8 @@
  * @file include/dfrn.php
  * @brief The implementation of the dfrn protocol
  *
- * https://github.com/friendica/friendica/wiki/Protocol
+ * @see https://github.com/friendica/friendica/wiki/Protocol and
+ * https://github.com/friendica/friendica/blob/master/spec/dfrn2.pdf
```
Comment (annando) [inline include/dfrn.php]: We could address this link relatively since it is available locally.

File: mod/dfrn_confirm.php [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
  *   1. A form was submitted by our user approving a friendship that originated elsewhere.
  *      This may also be called from dfrn_request to automatically approve a friendship.
  *
  *   2. We may be the target or other side of the conversation to scenario 1, and will
  *      interact with that process on our own user's behalf.
- *
+ *.
+ *  @see PDF with dfrn specs: https://github.com/friendica/friendica/blob/master/spec/dfrn2.pdf
+ *    You also find a graphic which describes the confirmation process at
+ *    https://github.com/friendica/friendica/blob/master/spec/dfrn2_contact_confirmation.png
```
Comment (annando) [inline mod/dfrn_confirm.php]: Please use relative addresses here as well.

**Pull request merged**